modules/openideal_user/openideal_user.module

```diff
@@ -90,14 +95,34 @@ function openideal_user_preprocess_page(&$variables) {
  * Implements hook_ENTITY_TYPE_insert().
  */
 function openideal_user_group_content_insert(GroupContent $entity) {
-  $event = new OpenidealUserGroupEvent($entity);
-  Drupal::service('event_dispatcher')->dispatch(OpenidealUserEvents::OPENIDEA_USER_JOINED_GROUP, $event);
+  if ($entity->getGroupContentType()->id() == 'idea-group_membership') {
+    $event = new OpenidealUserGroupEvent($entity);
+    \Drupal::service('event_dispatcher')->dispatch(OpenidealUserEvents::OPENIDEA_USER_JOINED_GROUP, $event);
+  }
 }
 
 /**
  * Implements hook_ENTITY_TYPE_delete().
  */
 function openideal_user_group_content_delete(GroupContent $entity) {
-  $event = new OpenidealUserGroupEvent($entity);
-  Drupal::service('event_dispatcher')->dispatch(OpenidealUserEvents::OPENIDEA_USER_LEFT_GROUP, $event);
+  if ($entity->getGroupContentType()->id() == 'idea-group_membership') {
+    $event = new OpenidealUserGroupEvent($entity);
+    \Drupal::service('event_dispatcher')->dispatch(OpenidealUserEvents::OPENIDEA_USER_LEFT_GROUP, $event);
+  }
+}
+
+/**
+ * Implements hook_flag_action_access().
+ *
+ * If user is Idea creator then restrict "unfollow" access.
+ */
+function openideal_idea_flag_action_access($action, FlagInterface $flag, AccountInterface $account, EntityInterface $flaggable = NULL) {
+  if ($flaggable
+    && $flaggable instanceof NodeInterface
+    && $action == 'unflag'
+    && $flaggable->bundle() == 'idea'
+    && $flaggable->uid->target_id == $account->id()) {
+    return AccessResult::forbidden();
+  }
+  return AccessResult::allowed();
```

Comment:
If there is no need to restrict or allow some specific access, we have to use `AccessResult::neutral()`.